src/selectors.js

```diff
@@ -0,0 +1,118 @@
+export const getSelectedCategoricalColumns = createSelector(
+  [getCategoricalColumns, getSelectedFeatures, getLabelColumn],
+  (categoricalColumns, selectedFeatures, labelColumn) => {
+    return categoricalColumns.filter(
+      column => (selectedFeatures.includes(column) || column === labelColumn)
+    );
+  }
+)
+
+export const getSelectedCategoricalFeatures = createSelector(
+  [getCategoricalColumns, getSelectedFeatures],
+  (categoricalColumns, selectedFeatures) => {
+    return categoricalColumns.filter(
+      column => selectedFeatures.includes(column)
+    );
+  }
+)
+
+export const getNumericalColumns = createSelector(
+  [getColumnsByDataType],
+  (columnsByDataType) => {
+    return filterColumnsByType(columnsByDataType, ColumnTypes.NUMERICAL);
+  }
+)
+
+export const getSelectedNumericalColumns = createSelector(
+  [getNumericalColumns, getSelectedFeatures, getLabelColumn],
+  (numericalColumns, selectedFeatures, labelColumn) => {
+    return numericalColumns.filter(
+      column => (selectedFeatures.includes(column) || column === labelColumn)
+    )
+  }
+)
+
+export const getSelectedNumericalFeatures = createSelector(
+  [getNumericalColumns, getSelectedFeatures],
+  (numericalColumns, selectedFeatures) => {
+    return numericalColumns.filter(
+      column => selectedFeatures.includes(column)
+    );
+  }
+)
+
+export const getUniqueOptionsByColumn = createSelector(
+  [getSelectedCategoricalColumns, getData],
+  (selectedCategoricalColumns, data) => {
+    let uniqueOptionsByColumn = {};
+    selectedCategoricalColumns.map(column => (
+      uniqueOptionsByColumn[column] = getUniqueOptions(data, column).sort()
+    ))
+    return uniqueOptionsByColumn;
+  }
+)
+
+export const getUniqueOptionsLabelColumn = createSelector(
+  [getLabelColumn, getData],
+  (labelColumn, data) => {
+    return getUniqueOptions(data, labelColumn).sort()
+  }
+)
+
+export const getUniqueOptionsCurrentColumn = createSelector(
+  [getCurrentColumn, getData],
+  (currentColumn, data) => {
+    return getUniqueOptions(data, currentColumn).sort()
+  }
+)
+
+export const getOptionFrequenciesCurrentColumn = createSelector(
+  [getCurrentColumn, getData],
+  (currentColumn, data) => {
+    let optionFrequencies = {};
+    for (let row of data) {
+      if (optionFrequencies[row[currentColumn]]) {
+        optionFrequencies[row[currentColumn]]++;
+      } else {
+        optionFrequencies[row[currentColumn]] = 1;
+      }
+    }
+    return optionFrequencies;
+  }
+)
+
+export const getExtremaByColumn = createSelector(
+  [getSelectedNumericalColumns, getData],
+  (getSelectedNumericalColumns, data) => {
+    let extremaByColumn = {};
+    getSelectedNumericalColumns.map(column => (
+      extremaByColumn[column] = getExtrema(data, column)
+    ))
+    return extremaByColumn;
+  }
+)
+
+export const getExtremaCurrentColumn = createSelector(
+  [getCurrentColumn, getData],
+  (currentColumn, data) => {
+    return getExtrema(data, currentColumn);
+  }
+)
```

Comment:
I guess at some point we might have multiple files of selectors, grouped by category?  A single file seems fine for now, though.

Reply:
Yeah, I imagine once I have the set of selectors written we might want to break them up kind of like I did for the redux file, but I'll do that in a future PR.